u8timeseries/utils/timeseries_generation.py

```diff
@@ -241,3 +242,44 @@ def holiday_timeseries(country_code: str,
     values = times.isin(scoped_country_holidays).astype(int)
 
     return TimeSeries.from_times_and_values(times, values)
+
+
+def upsample_timeseries(ts: TimeSeries, freq: str, method: str = 'pad') -> TimeSeries:
```

Comment:
Cool function! But maybe we could place it somewhere else? Because all other functions in timeseries_generation.py generate a new time series from scratch, so I'm not sure this fits. How about having this function as a method of TimeSeries (possibly static)?

Reply:
Good point